backends/occa/ceed-occa-basis.c

```diff
@@ -90,10 +90,21 @@ static int CeedBasisBuildKernel(CeedBasis basis) {
   // Test if we can get file's status, if not revert to occa://ceed/*.okl ******
   struct stat buf;
   if (stat(oklPath, &buf)!=0) {
-    dbg("[CeedElemRestriction][Create] Could NOT stat this OKL file: %s",oklPath);
-    dbg("[CeedElemRestriction][Create] Reverting to occa://ceed/*.okl");
-    strcpy(oklPath,"occa://ceed/ceed-occa-basis.okl");
+    dbg("[CeedBasis][BK] Could NOT stat this OKL file: %s",oklPath);
+    dbg("[CeedBasis][BK] Reverting to occa://ceed/*.okl");
+    // Try to stat ceed-occa-basis.okl in occa cache
+    ierr=sprintf(oklPath,"%s/ceed/libraries/ceed/ceed-occa-basis.okl",
+            ceed_data->occa_cache_dir);
+    if (ierr<0) return CeedError(ceed, 1, "With occa_cache_dir basis");
+    if (stat(oklPath, &buf)!=0) {
+      dbg("[CeedBasis][BK] Could NOT stat in OCCA cache: %s",oklPath);
+      // reverting to libceed_dir
+      ierr=sprintf(oklPath,"%s/okl/ceed-occa-basis.okl",ceed_data->libceed_dir);
+      if (ierr<0) return CeedError(ceed, 1, "With libceed_dir basis");
+    }else
+      strcpy(oklPath,"occa://ceed/ceed-occa-basis.okl");
```

Comment:
Can this bundle of code be put into an internal API? I'm concerned that the duplication will lead to inconsistent searching behavior.

Reply:
I refactored this part, now basis, restrict & even qfunctions use the same API.